### DISCUSSION TITLE
Flask in zip function handler startup command configuration when using CDK

### DISCUSSION THREAD

**Discussion by tehClaytonRH** [awslabs/aws-lambda-web-adapter#417] [Q&A]
I've managed to follow the [Flask Zip](https://github.com/awslabs/aws-lambda-web-adapter/tree/main/examples/flask-zip) example and adapt it for use with CDK, however, it seems as though the resulting python CDK construct isn't configured to allow for startup scripts being passed as handlers. The construct seems to concatenate the passed `index` and `handler` values when configuring the lambda, so as a workaround I did the following:

1. Set the index to `run.py`
2. Set the handler to `sh` 
(in anticipation that the `.py` file extension would be stripped, and the index and handler would be concatenated to `run.sh`)
3. Add an empty `run.py` file so as to avoid the `Runtime` error that `run.py` is not found

I was able to get everything working - so my question is whether there is a better approach to setting this lambda up via CDK that doesn't seem so hacky.

Thanks!

Roughly my function definition:
```python
web_adapter_layer_arn = f"arn:aws:lambda:{self.region}:753240598075:layer:LambdaAdapterLayerX86:20"
web_adapter_layer = aws_lambda_python.PythonLayerVersion.from_layer_version_arn(self, "web-adapter-layer", web_adapter_layer_arn)

self.handler = aws_lambda_python.PythonFunction(self, "FlaskApp",
      runtime=aws_lambda.Runtime.PYTHON_3_10,
      entry=src_dir,
      index='run.py',
      handler='sh',
      retry_attempts=2,
      timeout=Duration.seconds(30),
      vpc=vpc,
      environment={
          'AWS_LAMBDA_EXEC_WRAPPER': '/opt/bootstrap',
      },
      security_groups=[sg],
      layers=[web_adapter_layer],
)
```

**Comment by bnusunny** [comment]
PythonFunction construct tries to make it easy for developers to use it.  

NodejsFunction construct also had the same issue. It was fixed in [this issue](https://github.com/aws/aws-cdk/issues/24403). 

You could raise this issue for PythonFunction in CDK repo.
